modin/pandas/dataframe.py

```diff
@@ -111,9 +111,17 @@ def __init__(
                 "Distributing {} object. This may take some time.".format(type(data))
             )
             if is_list_like(data) and not is_dict_like(data):
-                data = [
+                old_dtype = getattr(data, "dtype", None)
+                values = [
                     obj._to_pandas() if isinstance(obj, Series) else obj for obj in data
                 ]
+                if isinstance(data, np.ndarray):
+                    data = np.array(values, dtype=old_dtype)
+                else:
+                    try:
+                        data = type(data)(values, dtype=old_dtype)
+                    except Exception:
```

Comment:
When would an Exception get thrown here?

Reply:
Generic `Exception` catching is maybe not necessarily what we want.

Reply:
An exception may be thrown if `type(data)` is a class that does not support `dtype` information. For example if `data` is `list`.

You're right that generic `Exception` is not suitable exception type, changed it to `TypeError` at the last commit